src/Lorentz/Contracts/RegistryDAO.hs

```diff
@@ -139,6 +145,61 @@ decisionLambda = do
         swap; toField #cpMaxProposalSize; ifSome (setField #ceMaxProposalSize) nop
         stSetField #sExtra
         nil
+    , #cUpdateReceiversType /->
+        caseT
+          ( #cAddReceivers /-> do
+              dip drop
+
+              stackType @([Address] ': store : _)
+              duupX @2; stToField #sExtra
+
+              stackType @(RegistryDaoContractExtra k v : [Address] ': store : _)
+              dup
+
+              stackType @(RegistryDaoContractExtra k v : RegistryDaoContractExtra k v : [Address] ': store : _)
+              toField #ceProposalReceivers
+
+              stackType @(Set Address : RegistryDaoContractExtra k v : [Address] ': store : _)
+              dip swap
+
+              stackType @(Set Address : [Address] : RegistryDaoContractExtra k v  ': store : _)
+              swap
+              stackType @([Address] : Set Address : RegistryDaoContractExtra k v  ': store : _)
+              iter $ do
+                dip $ push True
+                update
+
+              stackType @(Set Address ': RegistryDaoContractExtra k v : store : _)
+              setField #ceProposalReceivers
+              stSetField #sExtra
+              nil
+          , #cRemoveReceivers /-> do
+              dip drop
+
+              stackType @([Address] ': store : _)
+              duupX @2; stToField #sExtra
+
+              stackType @(RegistryDaoContractExtra k v : [Address] ': store : _)
+              dup
+
+              stackType @(RegistryDaoContractExtra k v : RegistryDaoContractExtra k v : [Address] ': store : _)
+              toField #ceProposalReceivers
+
+              stackType @(Set Address : RegistryDaoContractExtra k v : [Address] ': store : _)
+              dip swap
+
+              stackType @(Set Address : [Address] : RegistryDaoContractExtra k v  ': store : _)
+              swap
+              stackType @([Address] : Set Address : RegistryDaoContractExtra k v  ': store : _)
+              iter $ do
+                dip $ push False
```

Comment:
@sras regarding the origination size, it seems that the contract is close to the limit, so probably not much is needed to make it fit.

The first optimization possible is here: this `push` is the only difference between the two branches.
So I suggest you just add a `Bool` and an `[Address]` on the stack in the two case clauses and move the majority of code outside of it. This will cut the code size and may be enough to fit into the limits.

P.s. The same should be done in the Ligo version BTW

Reply:
This reduced the size quite a lot, but appears that we still need to reduce ~40 bytes more. I ll check it.